src/components/LayoutTree.tsx

```diff
@@ -540,50 +539,18 @@ const TreeNode = ({
         </div>
       </FadeTransition>
 
-      {/* DropEnd (cliff) */}
       {dragInProgress &&
-        cliff < 0 &&
         // do not render hidden cliffs
         // rough autofocus estimate
-        autofocusDepth - depth < 2 &&
-        Array(-cliff)
-          .fill(0)
-          .map((x, i) => {
-            const pathEnd = -(cliff + i) < path.length ? (path.slice(0, cliff + i) as Path) : HOME_PATH
-            const cliffDepth = unroot(pathEnd).length
-
-            // After table col2, shift the DropEnd left by the width of col1.
-            // This correctly positions the drop target for dropping after the table view.
-            // Otherwise it would be too far to the right.
-            const dropEndMarginLeft =
-              isTableCol2 && cliffDepth - depth < 0 ? treeThoughtsPositioned[index - 1].width || 0 : 0
-
-            return (
-              <div
-                key={'DropEnd-' + head(pathEnd)}
-                className={css({
-                  position: 'relative',
-                  top: '-0.2em',
-                  transition: `left {durations.fast} ease-out`,
-                  zIndex: 'subthoughtsDropEnd',
-                })}
-                style={{
-                  left: `calc(${cliffDepth - depth}em - ${dropEndMarginLeft}px + ${isTouch ? -1 : 1}px)`,
-                }}
-              >
-                <DropEnd
-                  depth={pathEnd.length}
-                  path={pathEnd}
-                  cliff={cliff}
-                  isLastVisible={isLastVisible}
-                  // Extend the click area of the drop target when there is nothing below.
-                  // The last visible drop-end will always be a dimmed thought at distance 1 (an uncle).
-                  // Dimmed thoughts at distance 0 should not be extended, as they are dimmed siblings and sibling descendants that have thoughts below
-                  // last={!nextChildId}
-                />
-              </div>
-            )
-          })}
+        autofocusDepth - depth < 2 && (
+          <DropCliff
+            cliff={cliff}
+            depth={depth}
+            path={path}
```

Comment:
these feel ok drill down, and none of them can't actually be extracted out of the loop

Reply:
Similar to [the other thread](https://github.com/cybersemics/em/pull/2785#discussion_r1927396150), I would _generally_ like to avoid prop drilling but consider it appropriate in this particular case 👍